source/pom.xml

```diff
@@ -185,18 +185,55 @@
         <plugin>
           <groupId>org.jacoco</groupId>
           <artifactId>jacoco-maven-plugin</artifactId>
-          <version>0.8.6</version>
+          <version>0.8.8</version>
         </plugin>
 
         <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-surefire-plugin</artifactId>
           <version>2.22.2</version>
         </plugin>
 
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-clean-plugin</artifactId>
+          <version>3.2.0</version>
+        </plugin>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.10.1</version>
+        </plugin>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-deploy-plugin</artifactId>
+          <version>3.0.0</version>
+        </plugin>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-install-plugin</artifactId>
+          <version>3.0.1</version>
+        </plugin>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-resources-plugin</artifactId>
+          <version>3.3.0</version>
+        </plugin>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-site-plugin</artifactId>
+          <version>3.12.1</version>
+        </plugin>
+
```

Comment:
What's the reasoning behind this? Iirc Maven will use the latest release version of the plugins if nothing is specified.

Reply:
Reproducability of a build: If you don't specify these, you'll have to work out what version of the plugins was used at the time of building a jar if you want to make an identical jar, which could be pretty hard.

If not specified I think the version depends on the super pom (at least that's what the versions-plugin indicates), which could depend on your maven installation? Not too sure that's guaranteed to always be the latest. For these basic plugins that shouldn't be that big of an issue, as I doubt they'll introduce breaking changes any time soon, so I guess we could stick to not specifying these plugins if there are good reasons to do that.

Reply:
Yes, if I look at the latest Maven docs they seem to even advise this.